### ISSUE TITLE
Support mutable maps

### ISSUE THREAD

**Issue by tomrf1** [circe/circe#555]
As discussed [here](https://github.com/guardian/fezziwig/issues/3#issuecomment-274151152), it would be useful to support encoding/decoding of `mutable.Map` as well as `immutable.Map`.

**Comment by tomrf1** [comment]
I’ve got something that appears to work in a fork [here](https://github.com/circe/circe/compare/master...tomrf1:tomrf1/support-mutable-map).

However, since `collection.Map` is invariant in its type parameter for the value, I have had to remove the `+` from the instances’ `V` type param. Does anyone know if this will cause any problems, i.e. will we need separate instances instead?

**Comment by travisbrown** [comment]
This is now available in master via #671:

```scala
scala> import scala.collection.mutable.HashMap
import scala.collection.mutable.HashMap

scala> import io.circe.Decoder, io.circe.syntax._
import io.circe.Decoder
import io.circe.syntax._

scala> val json = HashMap("a" -> 1, "b" -> 2).asJson
json: io.circe.Json =
{
  "b" : 2,
  "a" : 1
}

scala> Decoder[HashMap[String, Int]].decodeJson(json)
res0: io.circe.Decoder.Result[scala.collection.mutable.HashMap[String,Int]] = Right(Map(b -> 2, a -> 1))
```

I've removed the covariance, which I think shouldn't affect normal usage. So far it hasn't caused problems for me—we'll see if people complain when the first milestone comes out. :smile:
